src/main/java/org/sonatype/nexus/repository/chef/internal/metadata/util/TgzParser.java

```diff
@@ -18,11 +18,13 @@ public class TgzParser {
 
     @Nullable
     public InputStream getFileFromInputStream(final InputStream is, String fileName) throws IOException {
+        // Match the sought filename only if it is in the root of the archive or one subfolder down
+        String fileNameRegex = "^[^/]*[/]*" + fileName + "$";
```

Comment:
Given `bar` as `fileName`, this regexp is going to match `foobar`

I think `"^(?:[^/]*/)?" + fileName + "$"` would be better

Reply:
Done, thanks. I also noticed a nasty bug, we add fileName to the regex but fileName contains a ".", added proper escaping of it (the replaceall looks weird but works as intended, it's due to the first argument being a regex and the second a regular string, and java also using "/" as escape char. I've tried out the new regex locally and it works well.